manager/controlapi/service.go

```diff
@@ -751,6 +752,10 @@ func (s *Server) UpdateService(ctx context.Context, request *api.UpdateServiceRe
 			return status.Errorf(codes.NotFound, "service %s not found", request.ServiceID)
 		}
 
+		if service.PendingDelete {
```

Comment:
I don't think it can. Sure, we could check whether it's pending deletion as soon as we retrieve the service object above, but we would still also need to re-check here, once we hold the write lock; otherwise it could have been marked for deletion in the meantime. Maybe it's worth adding a comment to that effect?

Reply:
(added a comment to this effect)